lib/src/impl_ffi/impl_ffi.digest.dart

```diff
@@ -62,39 +62,104 @@ abstract class _Hash implements Hash {
       return out.copy(size);
     });
   }
+
+  @override
+  String hmacJwkAlg(HashImpl hash) {
+    if (hash == sha1) {
+      return 'HS1';
+    } else if (hash == sha256) {
+      return 'HS256';
+    } else if (hash == sha384) {
+      return 'HS384';
+    } else if (hash == sha512) {
+      return 'HS512';
+    } else {
+      throw UnsupportedError('hash is not supported');
+    }
+  }
```

Comment:
Not sure where to start:
 1. This shouldn't be declared in `impl_interface.digest.dart` because this is only used inside `impl_ffi/`, right? (or am I wrong?)
 2. Why does it take an argument? :rofl: It can just operate on `this`. If you wanted it to take an argument, it should have been a static method.
 3. Why not make it abstract, and then implement it on the classes?
```suggestion
  String get hmacJwkAlg;
```

Reply:
Yup, it's only being used in `impl_ffi/`. I did write some vestigial functions in `impl/js`. Performed a cleanup now. All tests pass on local.